ios/Classes/Blemulator/SimulatedAdapter.m

```diff
@@ -190,6 +207,14 @@ - (void)characteristicsForDevice:(NSString * _Nonnull)deviceIdentifier
 - (void)characteristicsForService:(double)serviceIdentifier
                           resolve:(NS_NOESCAPE Resolve)resolve
                            reject:(NS_NOESCAPE Reject)reject {
+    // TODO: - Error handling
```

Comment:
Check if device is connected and throw if not

Reply:
iOS's MBA does not check if device is connected in characteristicForService, but it rejects the call if service was not found. Addressed this accordingly